### ISSUE TITLE
Maximum call stack size exceeded when opening large document

### ISSUE THREAD

**Issue by rejhgadellaabsl** [ckeditor/ckeditor5#15120]
## 📝 Provide detailed reproduction steps (if any)

1. Open the html file (attached, see below) and copy the contents
2. . Paste it in an empty CKEditor

Note: We ran into this issue in Drupal 9.5.10, not entirely sure which version of CKEditor that ships?

### ✔️ Expected result

The content should be displayed

### ❌ Actual result

The page hangs for several seconds and then CKEditor crashes, the following error is logged to console:

```
Uncaught RangeError: Maximum call stack size exceeded
    at f.next (js_VaaKAlTD75yutATkQ0pAXOdN6Ms30ZsXy7jWB93uXlU.js:468:25123)
    at f.next (js_VaaKAlTD75yutATkQ0pAXOdN6Ms30ZsXy7jWB93uXlU.js:468:25860)
    at f._advanceToNextRow (js_VaaKAlTD75yutATkQ0pAXOdN6Ms30ZsXy7jWB93uXlU.js:468:26020)
    at f.next (js_VaaKAlTD75yutATkQ0pAXOdN6Ms30ZsXy7jWB93uXlU.js:468:25539)
    at f.next (js_VaaKAlTD75yutATkQ0pAXOdN6Ms30ZsXy7jWB93uXlU.js:468:25860)
    at f.next (js_VaaKAlTD75yutATkQ0pAXOdN6Ms30ZsXy7jWB93uXlU.js:468:25860)
    at f._advanceToNextRow (js_VaaKAlTD75yutATkQ0pAXOdN6Ms30ZsXy7jWB93uXlU.js:468:26020)
    at f.next (js_VaaKAlTD75yutATkQ0pAXOdN6Ms30ZsXy7jWB93uXlU.js:468:25539)
    at f.next (js_VaaKAlTD75yutATkQ0pAXOdN6Ms30ZsXy7jWB93uXlU.js:468:25860)
    at f.next (js_VaaKAlTD75yutATkQ0pAXOdN6Ms30ZsXy7jWB93uXlU.js:468:25860)
```

## 📃 Other details

* Browser: Chrome 117
* OS: Windows
* First affected CKEditor version: -
* Installed CKEditor plugins: -

As mention above, I'm not sure which version of CKEditor ships with Drupal 9.5.10. We do not have any additional plugins installed. The CKEditor config in Drupal is the default "Full HTML editor"

Example of content that crashes CKEditor:
[uittr.zip](https://github.com/ckeditor/ckeditor5/files/12792603/uittr.zip)

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by Witoso** [comment]
Hi, AFAICS the content is a large table. This issue was recently resolved in the [v39.0.2](https://github.com/ckeditor/ckeditor5/releases/tag/v39.0.2), #14785. I am not sure if this could be back ported to your Drupal version. I would reach out to the Drupal community.

**Comment by rejhgadellaabsl** [comment]
Thank you! For anyone running into this:

Looks like Drupal 9.5 will not receive the 39.x upgrade due to backward compat issues, but Drupal 10.x does:
- D10: https://www.drupal.org/project/drupal/issues/3380637#comment-15194396
- D9: https://www.drupal.org/project/drupal/issues/3380637#comment-15211223

Note: the links above are for CKEditor 39.0.1, so that doesn't actually include the patch for this issue, but I think it's safe to assume they will include it in a future patch version.
